DEV_FAQ.md

```diff
@@ -0,0 +1,94 @@
+
+Also double check that you're editing the source file, under `corehq/apps/<app_name>/static`, not a file in `staticfiles`
+
+> Web Apps and/or App Preview aren't working at all
+
+Formplayer isn't running properly. See [Formplayer](https://github.com/dimagi/commcare-hq/blob/master/DEV_FAQ.md#formplayer) below.
+
+> My web server is throwing a slew of web socket errors
+
+It happens, try restarting.
+
+## Generating Sample Data
+
+### SMS
+
+Check out the `generate_fake_sms_data` command.
+
+### Applications
+
+If you have an application on commcarehq.org, follow [these instructions](https://confluence.dimagi.com/display/commcarepublic/Copying+an+Application+between+Projects+or+Servers) to copy it to your local environment.
+
+There are also [three template apps](https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/app_manager/static/app_manager/template_apps) checked into the codebase.
+You can run [load_app_from_slug](https://github.com/dimagi/commcare-hq/blob/6021df8639dc0053c8dbdbb8690993be708776c5/corehq/apps/app_manager/views/apps.py#L510) in a django shell to import one of these apps. Note that you may wish to only run the first few lins of `load_app_from_slug` if you don't care about your app having multimedia.
+
+### Cases
+
+The easiest way to add cases locally is generally to use your local case importer. [Docs](https://confluence.dimagi.com/display/commcarepublic/Importing+Cases+Using+Excel)
+for this are extensive, but at its most basic, you can just upload a single-column file with a bunch of case names to create cases.
+
+### Cases + App + Data Dictionary
+
+The `create_case_fixtures` command will add a sample app, a data dictionary, and a set of relevant cases.
+
+## ElasticSearch
+
+ElasticSearch is the data source for reports, exports, and an assortment of parts of other features.
+Most issues with data not appearing locally, if you've already done something to create that data, are issues with ElasticSearch.
+
+You can run ElasticSearch continuously, mimicking a production environment, or on an as-needed basis. Either one of these can be simpler,
+depending on how much you're working with ES data.
+
+As described in the [dev setup guide](https://github.com/dimagi/commcare-hq/blob/master/DEV_SETUP.md#running-commcare-hq), you can
+use `run_ptop` to keep ES continually up to date.
+
+Alternatively, you can run `ptop_reindexer_v2` as needed to sync individual indexes. `ptop_reindexer_v2` works on one index at a time.
+Running it without any arguments will list the available indexes. The indexes you're most likely to use:
++ `sql-form` and `sql-case` populate form-based and case-based reports and exports
++ `user` and `group` may be useful, depending on what you're working on
++ `case_search` populates the Case list Explorer and Explore Case Data reports, as well as the case search/claim feature
```

Comment:
"for those domains with the `CASE_LIST_EXPLORER`, `EXPLORE_CASE_DATA` or `SEARCH_CLAIM` feature flags enabled"

Reply:
good point, https://github.com/dimagi/commcare-hq/pull/29895